### PR TITLE
Logger - PR 1 - send subscription state via event manager

### PR DESCRIPTION
**Pull request by mborof**
Sends subscription state changes via event manager so that setSubscriptionState doesn't need to directly reference propensity-server.

### CODE REVIEW

**Comment by mborof** [comment]
We discussed what to do about errors and we decided they should get logged to the console.  This is necessary to decouple sendSubscriptionState from PropensityServer so that we can have logger communicate it to PS without actually having PS passed to it (as you had requested)

**Comment by sohanirao** [comment]
My request is to simply hold off on this change. It has not been concluded that publishers dont need programmatic errors. We can come back to this. Logger can still communicate by sending event, yes. But this decoupling is not necessary since getPropensity() is only available in PropensityServer.

**Comment by mborof** [comment]
I will hold off on that for now

File: src/runtime/propensity-test.js [hunk 1]
```diff
@@ -151,16 +152,10 @@ describes.realWin('Propensity', {}, env => {
 
   it('should validate events sent to it and set appropriate defaults', () => {
     let hasError;
-    let receivedEvent;
-
```
Comment (sohanirao) [inline src/runtime/propensity-test.js]: i dont understand changes to this test file.

**Comment by sohanirao** [comment]
Changes to propensity server look good. Don't understand changes to propensity-test.

**Comment by mborof** [comment]
It was harder to undo the changes to PS than to just make a new one.
Please check this one now (612)

https://github.com/subscriptions-project/swg-js/pull/612/files

**Pull request closed without merging**